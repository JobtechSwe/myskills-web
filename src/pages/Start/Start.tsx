import React, { useContext } from 'react'
import { Context } from '../../Context'
import { RouteComponentProps, Link } from '@reach/router'
import styled from '@emotion/styled'

const Wrapper = styled.section`
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-flow: column nowrap;
`

const Start: React.FC<RouteComponentProps> = () => {
  const {
    state: { logoActive },
    dispatch,
  } = useContext(Context)

  return (
    <Wrapper>
      <div>
        <Link to="/login">Login</Link>
      </div>
      <div>
        <Link to="/skapa-cv">Skapa Cv</Link>
      </div>
    </Wrapper>
  )
}

export default Start
