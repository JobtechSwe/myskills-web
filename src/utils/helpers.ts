import Cookies from 'js-cookie'
import { LocalStateProps, ContactInformationProps } from '../graphql/client'
import {
  EducationInput,
  Occupation,
  OntologyRelationResponse,
} from '../generated/myskills'

export const getCookie = (name: string) => Cookies.get(name)

export const setCookie = (
  name: string,
  value: string,
  options: object = {}
) => {
  Cookies.set(name, value, options)
}

export const removeCookie = (name: string) => {
  Cookies.remove(name)
}

export const redirect = (route: string) => (location.href = route)

interface StorageEntryProps {
  type: string
}

interface Educations extends StorageEntryProps {
  type: 'educations'
  data: EducationInput[]
}

interface OccupationStorage extends StorageEntryProps {
  type: 'occupation'
  data: Occupation
}

interface Skills extends StorageEntryProps {
  type: 'skills'
  data: OntologyRelationResponse[]
}
interface Traits extends StorageEntryProps {
  type: 'traits'
  data: string[]
}
interface WhoAmI extends StorageEntryProps {
  type: 'whoAmI'
  data: string
}

interface ContactInformation extends StorageEntryProps {
  type: 'contact'
  data: ContactInformationProps
}

export type StorageEntry =
  | ContactInformation
  | Educations
  | OccupationStorage
  | Skills
  | Traits
  | WhoAmI

export const storageHelper = {
  load: (initialState: LocalStateProps) =>
    Object.entries(initialState).reduce(
      (acc, [prop, val]) => ({
        ...acc,
        [prop]: JSON.parse(localStorage.getItem(prop) as string) || val,
      }),
      {} as LocalStateProps
    ),
  set: (payload: StorageEntry) =>
    localStorage.setItem(payload.type, JSON.stringify(payload.data)),
}
