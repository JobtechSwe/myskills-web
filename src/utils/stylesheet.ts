const colors = {
  lassekongo: 'dodgerblue',
  primary: 'pink',
}

const sizes = ['0', '1rem', '2rem', '4rem']

const typography = {
  default: 'Arial',
}

const space = {
  none: sizes[0],
  small: sizes[1],
  medium: sizes[2],
  big: sizes[3],
}

export default { colors, space, typography }
