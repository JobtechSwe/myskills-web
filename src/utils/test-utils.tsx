import * as React from 'react'
import ApolloClient from 'apollo-client'
import { ApolloProvider } from 'react-apollo-hooks'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { MockLink, MockedProvider } from 'react-apollo/test-utils'
import { ThemeProvider } from 'emotion-theming'
import { render as rtlRender } from 'react-testing-library'
import { theme } from 'theme'

export const render = (children: React.ReactNode, mocks?: any[]) => {
  const client = new ApolloClient({
    cache: new InMemoryCache({ addTypename: true }),
    link: new MockLink(mocks || []),
  })

  return rtlRender(
    <MockedProvider addTypename={true} mocks={mocks}>
      <ApolloProvider client={client}>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </ApolloProvider>
    </MockedProvider>
  )
}
