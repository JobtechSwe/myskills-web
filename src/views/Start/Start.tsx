import React from 'react'
import Flex from '../../components/Flex'
import { RouteComponentProps } from '@reach/router'
import Button from '../../components/Button'
import { InternalLink } from '../../components/Link'

const Start: React.FC<RouteComponentProps> = () => {
  return (
    <Flex alignItems="center" flexDirection="column" justifyContent="center">
      <Flex p={32}>
        <InternalLink to="/skapa-cv">
          <Button variant="primary">Skapa Cv</Button>
        </InternalLink>
      </Flex>
      <Flex alignItems="center" flexDirection="column">
        <InternalLink mb={12} to="/login">
          Login
        </InternalLink>
        <InternalLink to="/register">Register</InternalLink>
      </Flex>
    </Flex>
  )
}

export default Start
