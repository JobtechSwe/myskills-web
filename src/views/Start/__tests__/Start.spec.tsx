import * as React from 'react'
import Start from '../Start'
import { render } from '../../../utils/test-utils'
import { wait } from 'react-testing-library'

describe('views/Start', () => {
  it('renders Start', async () => {
    const { container } = render(<Start />)

    await wait()

    expect(container).toMatchSnapshot()
  })
})
