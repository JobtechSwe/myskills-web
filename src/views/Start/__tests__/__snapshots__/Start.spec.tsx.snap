// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/Start renders Start 1`] = `
<div>
  <div
    class="css-4ia2ti"
  >
    <div
      class="css-owzlzk"
    >
      <img
        alt="logo"
        src="/images/logo.svg"
      />
      <hr
        class="css-15g94ap"
      />
      <p
        class="css-qt7uhv"
        color="persianBlue"
        font-family="default"
        font-size="small"
      >
        Ditt digitala CV
      </p>
      <img
        alt="clouds"
        class="css-smyzqs"
        src="/images/clouds.svg"
      />
      <a
        class="css-o0whc9"
        color="persianBlue"
        font-weight="500"
        href="/skapa-cv"
        mt="30"
        width="1"
      >
        <button
          class="css-kun9in"
          font-family="default"
          font-size="medium"
          width="1"
        >
          Skapa CV
        </button>
      </a>
      <a
        class="css-19sxhvi"
        color="persianBlue"
        font-weight="500"
        href="/login"
        mt="10"
        width="1"
      >
        <button
          class="css-bthv9p"
          font-family="default"
          font-size="medium"
          width="1"
        >
          Logga in med 
          <span
            class="css-1x7kjku"
            font-family="default"
            font-size="medium"
            font-weight="700"
          >
            Egendata
          </span>
        </button>
      </a>
      <div
        class="css-efe99a"
      >
        <img
          alt="info icon"
          src="/images/info.svg"
        />
        <p
          class="css-dnci7j"
          font-family="default"
          font-size="small"
        >
          Med Egendata får du kontroll över din data.
           
          <a
            class="css-bqii04"
            rel="noopener"
            target="_blank"
            text-decoration="underline"
          >
            Läs mer om hur det fungerar
          </a>
        </p>
      </div>
    </div>
  </div>
  <div
    class="css-bfkqop"
  >
    <h2
      class="css-54k13x"
      color="cloudBurst"
      font-family="default"
      font-size="24"
    >
      Hur fungerar det?
    </h2>
    <div>
      <h3
        class="css-1vyeit5"
        color="cloudBurst"
        font-family="default"
        font-size="16"
        font-weight="bold"
      >
        Skapa din profil
      </h3>
      <p
        class="css-1el412k"
        color="cloudBurst"
        font-family="default"
        font-size="small"
      >
        Svara på frågor om dig själv och skapa ditt CV.
      </p>
    </div>
    <div>
      <h3
        class="css-1vyeit5"
        color="cloudBurst"
        font-family="default"
        font-size="16"
        font-weight="bold"
      >
        Lagra ditt CV
      </h3>
      <p
        class="css-1g7mhfo"
        color="cloudBurst"
        font-family="default"
        font-size="small"
      >
        Genom Egendata väljer du själv var ditt CV ska sparas.
      </p>
    </div>
    <div>
      <h3
        class="css-1vyeit5"
        color="cloudBurst"
        font-family="default"
        font-size="16"
        font-weight="bold"
      >
        Spara tid
      </h3>
      <p
        class="css-1g7mhfo"
        color="cloudBurst"
        font-family="default"
        font-size="small"
      >
        Hämta och dela din information istället för att skapa ett nytt CV för varje jobbsökarsajt.
      </p>
    </div>
  </div>
</div>
`;
